src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/StorePal.cs

```diff
@@ -32,27 +32,47 @@ public static ILoaderPal FromBlob(byte[] rawData, SafePasswordHandle password, X
 
             X509ContentType contentType = X509Certificate2.GetCertContentType(rawData);
 
-            SafeTemporaryKeychainHandle tmpKeychain;
+            SafeKeychainHandle keychain;
+            bool exportable = true;
 
             if (contentType == X509ContentType.Pkcs12)
             {
-                tmpKeychain = Interop.AppleCrypto.CreateTemporaryKeychain();
+                if ((keyStorageFlags & X509KeyStorageFlags.EphemeralKeySet) == X509KeyStorageFlags.EphemeralKeySet)
+                {
+                    throw new PlatformNotSupportedException(SR.Cryptography_X509_NoEphemeralPfx);
+                }
+
+                exportable = (keyStorageFlags & X509KeyStorageFlags.Exportable) == X509KeyStorageFlags.Exportable;
+
+                bool persist =
+                    (keyStorageFlags & X509KeyStorageFlags.PersistKeySet) == X509KeyStorageFlags.PersistKeySet;
+
+                keychain = persist
+                    ? Interop.AppleCrypto.SecKeychainCopyDefault()
+                    : Interop.AppleCrypto.CreateTemporaryKeychain();
             }
             else
             {
-                tmpKeychain = SafeTemporaryKeychainHandle.InvalidHandle;
+                keychain = SafeTemporaryKeychainHandle.InvalidHandle;
                 password = SafePasswordHandle.InvalidHandle;
             }
 
             // Only dispose tmpKeychain on the exception path, otherwise it's managed by AppleCertLoader.
             try
             {
-                SafeCFArrayHandle certs = Interop.AppleCrypto.X509ImportCollection(rawData, contentType, password, tmpKeychain);
-                return new AppleCertLoader(certs, tmpKeychain);
+                SafeCFArrayHandle certs = Interop.AppleCrypto.X509ImportCollection(
+                    rawData,
+                    contentType,
+                    password,
+                    keychain,
+                    exportable);
+
+                // If the default keychain was used, null will be passed to the loader.
+                return new AppleCertLoader(certs, keychain as SafeTemporaryKeychainHandle);
             }
             catch
             {
-                tmpKeychain.Dispose();
+                keychain.Dispose();
```

Comment:
I assume no problems will be caused by disposing of the singleton InvalidHandle?